bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -329,10 +346,222 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
             this._sandbox.notifyAll(evt);
         },
+        /**
+         * Dalete layer group
+         * @method deleteLayerGroup
+         * @param  {Object}         data group data
+         */
+        deleteLayerGroup: function(data) {
+            var me = this;
+            var newGroups = [];
+            if (data.parentId === -1) {
+                me._layerGroups.forEach(function(group) {
+                    if (group.id != data.id) {
+                        newGroups.push(group);
+                    }
+                });
+                me._layerGroups = newGroups;
+            } else {
+                var editable = me.getAllLayerGroups(data.parentId);
+                editable.groups.forEach(function(group) {
+                    if (group.id != data.id) {
+                        newGroups.push(group);
+                    }
+                });
+                editable.groups = newGroups;
+            }
+        },
+        /**
+         * Updata layer groups
+         * @method updateLayerGroups
+         * @param  {Object}          data data
+         */
+        updateLayerGroups: function(data) {
+            var me = this;
+            var editable = me.getAllLayerGroups(data.id);
+            // if found then update only
+            if (editable) {
+                editable.name = data.name;
+                editable.parentId = data.parentId;
+                editable.selectable = data.selectable;
+            } else if (data.parentId === -1) {
+                me.getAllLayerGroups().push({
+                    groups: [],
+                    id: data.id,
+                    name: data.name,
+                    layers: [],
+                    parentId: data.parentId,
+                    selectable: data.selectable
+                });
+            } else {
+                me.getAllLayerGroups(data.parentId).groups.push({
+                    groups: [],
+                    id: data.id,
+                    name: data.name,
+                    layers: [],
+                    parentId: data.parentId,
+                    selectable: data.selectable
+                });
+            }
+        },
+
+        /**
+         * Update layer groups
+         * @method updateGroupsLayers
+         * @param  {String}          layerId      layerid
+         * @param  {Object}          newLayerConf new layer conf
+         * @param  {Boolean}         deleteLayer delete layer
+         * @param  {Boolean}         newLayer is new layer
+         */
+        updateGroupsLayers: function(layerId, newLayerConf, deleteLayer, newLayer) {
+            var me = this;
+            var groups = [];
+            var group = null;
+
+            if (me._layerGroups.length === 0) {
+                return;
+            }
+
+            // update or insert
+            if (!deleteLayer) {
+                var isLayerInGroup = function(groupLayers) {
```

Comment:
Rename isLayerInGroup to something like layerIndexInGroup as that's what the value is.